### DISCUSSION TITLE
Port Forwarding vs. VPN

### DISCUSSION THREAD

**Discussion by 68rich** [jamulussoftware/jamulus#3120] [Technical Support & Help]
Hi All,
I've been running a private jamulus server for a while now. I don't have a static IP address so I've been telling people the IP address on an as & when basis. I've used port forwarding to forward packets to the server. I have a number of OpenWrt routers scattered around the network to segregate different bits of my network. This has always worked really well for us in the past.
Along comes a new ISP with a gigabit of fibre connection, and now port forwarding doesn't happen any more.
I've been thinking about setting up a VPN and granting users access to it so they can see the server.
Has anyone got any suggestions about architecture for that?
My first thought was to set up an OpenVPN server on a spare OpenWrt router - but this is a learning experience for me so I thought I'd raise the question with everyone here before starting.

Any suggestions?

**Comment by pljones** [comment]
Are you running a Jamulus Server that registers with a Jamulus Directory on the wider internet from where the Server is hosted?

Are you using IPv6 exclusively?

**Reply by 68rich** [reply]
No - I'm not registering the server anywhere. It's not even powered on unless we need it for a rehearsal.
Also not exclusively using IPv6. Users have always connected using the IPv4 address in the past.

**Reply by ann0see** [reply]
> Users have always connected using the IPv4 address in the past.

I believe there might be a CGN: https://en.wikipedia.org/wiki/Carrier-grade_NAT

I'd register with a server to overcome this - or get a static IP.

**Reply by ann0see** [reply]
Maybe registering and then immediately unregistering if everyone is on board would work. See https://jamulus.io/wiki/Tips-Tricks-More#converting-a-registered-server-to-an-unregistered-one-on-the-fly

**Reply by pljones** [reply]
> Maybe registering and then immediately unregistering if everyone is on board would work. See https://jamulus.io/wiki/Tips-Tricks-More#converting-a-registered-server-to-an-unregistered-one-on-the-fly

I should update that to mention you don't need to be using the GUI Server -- it should work just as well with the headless Server.  In fact, you shouldn't need to wait for others to connect, so long as they have your IP/port and join within a few minutes of you registering successfully.  The ping back you receive from the Directory should keep your router primed for a while.  Once you're exchanging traffic with your Clients, the ports should stay open to allow the on-going exchange.

(Caveat emptor... I've a static IP and port mapping in place.)

> It's not even powered on unless we need it for a rehearsal.

Just to allay any concerns, you don't need to worry about being available all the time in order to register with a Directory.  The Public (built-in) Directories are open to _any_ Jamulus Server operator wanting to register for any amount of time for any reason.

**Reply by 68rich** [reply]
Hi Everyone - Thanks for your replies and sorry for being so quiet on this for a while. I have registered the server for the time being and its working well for us.
In the future I'll start looking at IPv6 routing which may be the way to go, but given that everything is working well now, I'm going to sit back and enjoy the jam.
